uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/request.controller.spec.js

```diff
@@ -85,6 +85,8 @@
         AbsenceTypeAPI = _AbsenceTypeAPI_;
         AbsencePeriodAPI = _AbsencePeriodAPI_;
         pubSub = _pubSub_;
+        notification = _notificationService_;
+        window.alert = function () {}; // prevent alert from being logged in console
```

Comment:
Since `notification.success` doesn't return anything, why not just spy on these methods without calling them through?

`spyOn(notification, 'success');`

In that way you don't need to remove the default behaviour of window.alert.

Reply:
If I didn't call through it, I would not have spotted a bug within unit tests when the notification service is called with null body. To be honest, I am not a fan of stopping functions. So I would prefer overriding alert instead.